### PR TITLE
Add `async` query parameter to control send endpoint behavior

### PR DESCRIPTION
**Pull request by Anthuang**
Open to naming suggestions for the query parameter.

Closes: #607

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
LGTM assuming it works. But should be reviewed by @ppatierno as the Bridge guru.

So nits:
* Should something be added to the docs?
* Should we add a note about this to the CHANGELOG.md file?

**Comment by Anthuang** [comment]
I will wait for @ppatierno's review, but I'm also testing it currently in a staging environment (have to also write the glue code).

I'll look into adding it to the docs/changelog as well

**Review by scholzj — COMMENTED** [review]
I think you need to rebase it as well. There seems to be some conlfict as you probably created it from some older `main` branch.

File: bin/docker/docker_setup.sh
Comment (scholzj) [inline bin/docker/docker_setup.sh:1]: I don't think this should be here.

File: config/bridge.properties
Comment (scholzj) [inline config/bridge.properties:1]: This file probably neither?

File: bin/docker/docker_setup.sh
Comment (Anthuang) [reply bin/docker/docker_setup.sh]: Oops let me delete it. Messed up while rebasing

**Review by ppatierno — CHANGES_REQUESTED** [review]

File: src/main/java/io/strimzi/kafka/bridge/http/HttpSourceBridgeEndpoint.java
Comment (Anthuang) [inline src/main/java/io/strimzi/kafka/bridge/http/HttpSourceBridgeEndpoint.java:228]: This is to reduce `Cyclomatic Complexity`

**Comment by Anthuang** [comment]
Not sure what's causing this error:

```
2022-05-31T16:07:13.3121820Z [ERROR] io.strimzi.kafka.bridge.http.AdminClientIT  Time elapsed: 30.916 s  <<< ERROR!
2022-05-31T16:07:13.3146277Z java.util.concurrent.TimeoutException: 
2022-05-31T16:07:13.3148525Z The test execution timed out. Make sure your asynchronous code includes calls to either VertxTestContext#completeNow(), VertxTestContext#failNow() or Checkpoint#flag()
```

**Comment by Anthuang** [comment]
That seems to have fixed it (although I'm not sure why), can someone approve the rest of the workflows.

**Comment by ppatierno** [comment]
@Anthuang I checked the last tests running with the error. I am not sure that your latest change (about closing) is the fix for the test not working even because the "async" related stuff is not used at all in the failing tests. I think you should revert back your latest change because as you said "That seems to have fixed it (although I'm not sure why)" and investigate the root cause of that test failure. It's possible there is some other subtle problem or the test is flaky.

**Comment by ppatierno** [comment]
My concern is that we don't have these tests failing on our main branch so I was wondering if further investigation was needed.

**Comment by scholzj** [comment]
Well, it seemed to have passed now without any changes to existing tests. So maybe it is fixed now? TBH, in #629 (where the change is actually not used during the unit tests at all) I had also quite a lot of failures and needed restarts :-/.

File: src/main/java/io/strimzi/kafka/bridge/http/HttpSourceBridgeEndpoint.java
Comment (ppatierno) [inline src/main/java/io/strimzi/kafka/bridge/http/HttpSourceBridgeEndpoint.java:69]: why this name? I mean, from the code inside, it doesn't show up that it's closing when there is an error.

Comment (Anthuang) [reply src/main/java/io/strimzi/kafka/bridge/http/HttpSourceBridgeEndpoint.java:69]: Do you have a suggestion for the name? I'm just trying to keep the `Cyclomatic Complexity` within the bounds without changing too much code with this PR.

Comment (ppatierno) [reply src/main/java/io/strimzi/kafka/bridge/http/HttpSourceBridgeEndpoint.java:69]: Let's try with something like `maybeClose` taking into account that the `closing` has to be validate.

**Review by ppatierno — CHANGES_REQUESTED** [review]

**Review by ppatierno — APPROVED** [review]
LGTM. Thanks! @scholzj would you mind to have another pass please?

**Review by scholzj — APPROVED** [review]
Thanks for the PR.

**Pull request merged**
